x-pack/plugins/observability_solution/infra/public/components/ml/anomaly_detection/anomalies_table/anomalies_table.tsx

```diff
@@ -134,13 +152,55 @@ const AnomalyActionMenu = ({
   ];
 
   if (!disableShowInInventory) {
-    items.push(
-      <EuiContextMenuItem key="showInInventory" icon="search" onClick={showInInventory}>
+    const buildFilter = buildCombinedAssetFilter({
+      field: HOST_NAME_FIELD,
+      values: influencers,
+      dataView: metricsView?.dataViewReference,
+    });
+
+    let newFilter: Filter[] = [];
+    if (!Array.isArray(buildFilter)) {
+      newFilter = [buildFilter];
+    }
+
+    const showInHostsItem = !hostName ? (
+      <EuiContextMenuItem
+        key="showAffectedHosts"
+        icon="search"
+        data-test-subj="infraAnomalyFlyoutShowAffectedHosts"
+        href={hostsLocator?.getRedirectUrl({
+          dateRange: {
+            from: timeRange.start,
+            to: timeRange.end,
+          },
+          filters: newFilter,
+        })}
+      >
         <FormattedMessage
-          id="xpack.infra.ml.anomalyFlyout.actions.showInInventory"
-          defaultMessage="Show in Inventory"
+          id="xpack.infra.ml.anomalyFlyout.actions.showAffectedHosts"
+          defaultMessage="Show affected Hosts"
         />
       </EuiContextMenuItem>
+    ) : (
+      <></>
```

Comment:
nit: do we need to render this fragment?
```suggestion
     null
```

Reply:
I did it initially, but I had to change to `<></>` because TS complains about `null`:
<img width="1647" alt="image" src="https://github.com/user-attachments/assets/dc95df07-3b6c-470b-b8e8-b6fc25668478">